pkg/typesystem/typesystem.go

```diff
@@ -491,9 +507,10 @@ func (t *TypeSystem) RecursiveUsersetCanFastPath(objectTypeRelation string, user
 		// this means the node cannot be found. The safe thing to do is to use the slow path.
 		return false
 	}
+	// FIXME: for now, we will disable fastpath for cases where publicly wildcard can be assigned to recursive userset.
 	return t.verifyNodeEdgesOptimizable(curAuthorizationModelNode,
 		expectedEdgeAndNodeType{expectedNodeForObjectRel: curAuthorizationModelNode, expectedEdgeType: graph.DirectEdge},
-		userType)
+		userType) && !t.hasPublicWildcardNode(curAuthorizationModelNode)
```

Comment:
should we skip the call to `verifyNodeEdgesOptimizable` all together in this case?

Reply:
flipped order of check so that hasPublicWildcardNode is checked first.